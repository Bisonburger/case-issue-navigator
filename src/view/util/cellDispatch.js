import React from "react";
import { ELIS_CASE_BASE_URL } from "../../controller/config";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

export const cellDispatch = {
  LINK: (r, rowData) => {
    if (!r) {
      return null;
    }

    return (
      <>
        <a href={ELIS_CASE_BASE_URL + r} target="_elis_viewer">
          {r}
        </a>
        {rowData && rowData.previouslySnoozed && (
          <React.Fragment>
            &nbsp;
            <FontAwesomeIcon
              icon="exclamation-triangle"
              className="text-accent-warm"
              aria-label="This case has been previously snoozed"
            />
          </React.Fragment>
        )}
      </>
    );
  },
  DATE: d => {
    const datum = new Date(d);

    if (d && !isNaN(datum)) {
      return (
        datum.getMonth() + 1 + "/" + datum.getDate() + "/" + datum.getFullYear()
      );
    }

    return "Invalid date";
  }
};
