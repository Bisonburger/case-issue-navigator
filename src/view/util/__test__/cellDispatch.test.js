import { cellDispatch } from "../cellDispatch";

describe("cellDispatch", () => {
  describe("LINK", () => {
    it("should return a valid link with no desnoozed notification", () => {
      expect(
        cellDispatch["LINK"]("FAK123", { desnoozed: false })
      ).toMatchSnapshot();
    });
    it("should return a valid link with a desnoozed notification", () => {
      expect(
        cellDispatch["LINK"]("FAK123", { desnoozed: true })
      ).toMatchSnapshot();
    });
    it("should handle return null for invalid inputs", () => {
      expect(cellDispatch["LINK"](null, null)).toBe(null);
    });
  });

  describe("DATE", () => {
    it("should produce a well-formatted date", () => {
      const date = "July 10, 2018";
      expect(cellDispatch["DATE"](date)).toBe("7/10/2018");
    });
    it("should do something with a bad input", () => {
      expect(cellDispatch["DATE"](null)).toBe("Invalid date");
      expect(cellDispatch["DATE"]("Hello world")).toBe("Invalid date");
    });
  });
});
