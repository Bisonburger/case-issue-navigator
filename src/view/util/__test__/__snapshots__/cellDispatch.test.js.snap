// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cellDispatch LINK should return a valid link with a desnoozed notification 1`] = `
<React.Fragment>
  <a
    href="https://internal-prod-elis2.uscis.dhs.gov/InternalApp/app/#/case/FAK123"
    target="_elis_viewer"
  >
    FAK123
  </a>
  <FontAwesomeIcon
    border={false}
    className="text-accent-warm"
    fixedWidth={false}
    flip={null}
    icon="exclamation-triangle"
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
</React.Fragment>
`;

exports[`cellDispatch LINK should return a valid link with no desnoozed notification 1`] = `
<React.Fragment>
  <a
    href="https://internal-prod-elis2.uscis.dhs.gov/InternalApp/app/#/case/FAK123"
    target="_elis_viewer"
  >
    FAK123
  </a>
</React.Fragment>
`;
